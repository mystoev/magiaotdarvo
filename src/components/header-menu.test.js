import React from 'react';
import { MemoryRouter as Router } from 'react-router';
import { renderWithTheme } from '../../test/utils';

import HeaderMenu from './header-menu';

describe('<HeaderMenu />', () => {
  it('should render', () => {
    const { container } = renderWithTheme(
      <Router>
        <HeaderMenu links={[]} />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render with links', () => {
    const links = [
      { to: '/path1', text: 'Link1' },
      { to: '/path2', text: 'Link2' }
    ];

    const { container, queryByText } = renderWithTheme(
      <Router>
        <HeaderMenu links={links} />
      </Router>
    );

    expect(container).toMatchSnapshot();
    links.forEach(({ text }) => {
      expect(queryByText(text.toUpperCase())).toBeInTheDocument();
    });
  });
});
