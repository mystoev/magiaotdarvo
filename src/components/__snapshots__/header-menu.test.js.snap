// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderMenu /> should render 1`] = `
<div>
  <div
    class="header-menu__MenuSmall-sc-1yxapcf-0 Oasdt"
  >
    <svg
      class="menu-icon"
    />
  </div>
  <ul
    class="header-menu__MenuDefault-sc-1yxapcf-1 fEbyIa"
  />
</div>
`;

exports[`<HeaderMenu /> should render with links 1`] = `
<div>
  <div
    class="header-menu__MenuSmall-sc-1yxapcf-0 Oasdt"
  >
    <svg
      class="menu-icon"
    />
  </div>
  <ul
    class="header-menu__MenuDefault-sc-1yxapcf-1 fEbyIa"
  >
    <li>
      <a
        href="/path1"
      >
        LINK1
      </a>
    </li>
    <li>
      <a
        href="/path2"
      >
        LINK2
      </a>
    </li>
  </ul>
</div>
`;
